code/smoothie.js

```diff
@@ -0,0 +1,314 @@
+let smoothie = {
+  mainFruit: undefined,
+  mixer: undefined,
+  superBooster: undefined
+}
+
+// Defining the favoriteFruit-variable in the global scope
+let favoriteFruit = undefined
+
+// Select a favorite fruit to make it appear more frequently in the mix
+const selectFavorite = (value) => {
+  event.preventDefault()
+  favoriteFruit = value.replace(/(^"|"$)/g, '')
+
+
+  // This is how I would solve the "Shuffle Favorite Weapon"-problem:
+  // (I would connect the function to an onclick event on a button, 
+  // and run that with an if-statement similar to the one below)
+
+  // const randomFavorite = () => {
+  //   favoriteFruit = randomSelector(mainFruits).name
+  //   console.log("Hello", favoriteFruit)
+  // }
+
+
+  // If a favorite is selected, remove the extra favorite-objects in array
+  const removeFavorites = () => {
+    if (mainFruitsPlusFavorite.length > 6) {
+      mainFruitsPlusFavorite.splice(-3, 3)
+    }
+  }
+
+  // If favorite is selected in browser, add matching object from mainFruit-array x 3 
+  // to a new array which will later be used to randomize fruit 
+  if (favoriteFruit == "Apple") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[0], mainFruits[0], mainFruits[0])
+  } else if (favoriteFruit == "Banana") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[1], mainFruits[1], mainFruits[1])
+  } else if (favoriteFruit == "Mango") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[2], mainFruits[2], mainFruits[2])
+  } else if (favoriteFruit == "Orange") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[3], mainFruits[3], mainFruits[3])
+  } else if (favoriteFruit == "Raspberry") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[4], mainFruits[4], mainFruits[4])
+  } else if (favoriteFruit == "Lemon") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[5], mainFruits[5], mainFruits[5])
+  }
+}
+
+// Function that selects a random object from chosen array
+const randomSelector = array => {
+  return array[Math.floor(Math.random() * array.length)]
+}
+
+// Function that toggles loading-gif
+const toggleLoader = (whatBox) => {
+  document.getElementById(whatBox).classList.add('loading')
+  // remove the loading-gif...
+  const removeLoader = () => {
+    document.getElementById(whatBox).classList.remove('loading')
+  }
+  //... after 1.5 second
+  setTimeout(removeLoader, 1500)
+}
+
+// 3 x Display random image and title from each category 
+// + add value to smoothie-object
+// + check if smoothie-button color should be changed
+const pickMainFruit = () => {
+  mainFruit = randomSelector(mainFruits)
+
+  // Pick a random fruit, unless favorite is selected 
+  // (then pick random from mainFruitsPlusFavorite-array)
+  if (mainFruitsPlusFavorite[6] == undefined) {
+    mainFruit = randomSelector(mainFruits)
+  } else {
+    mainFruit = randomSelector(mainFruitsPlusFavorite)
+  }
```

Comment:
This is where a `preferredFruit` variable would be handy:
instead of
```javascript
  if (mainFruitsPlusFavorite[6] == undefined) {
```
```javascript
  if (preferredFruit == undefined) {
```

Once again with the idea that the 6th element could change when adding more ingredients